References.php

```diff
@@ -222,4 +222,29 @@ public function interlace(string $interlaceType): string
             'http://www.imagemagick.org/script/command-line-options.php#interlace'
         ));
     }
+
+    /**
+     * Checks that threshold value is valid according to ImageMagick command line reference.
+     */
+    public function threshold(string $threshold): float
+    {
+        $threshold = \trim($threshold);
+
+        if (substr($threshold, -1) == '%') {
+            $percent_num = substr($threshold, 0, -1);
+            if (is_numeric($percent_num))
+                return $threshold;
+        } else {
+            if (is_int(filter_var($threshold, FILTER_VALIDATE_INT)) && intval($threshold) >= 0)
+                return $threshold;
+        }
+
+        throw new \InvalidArgumentException(\sprintf(
+            'The specified threshold parameter (%s) is invalid.'."\n".
+            'The value must be in percentage or an absolute integer'."\n".
```

Comment:
```suggestion
            'The value must be in percentage or an integer'."\n".
```

Reply:
My bad, I misunderstood that part in the docs that mention it has to be "absolute integer value corresponding to the desired channel value" and "as an integer, the minimum attainable value is 0". Yes, negative ints are supported, just tested.